politeiawww/proposals.go

```diff
@@ -838,6 +890,54 @@ func (p *politeiawww) processProposalDetails(propDetails www.ProposalsDetails, u
 	return &reply, nil
 }
 
+// processBatchProposals fetches a list of proposals from the records
+// cache and returns them
+func (p *politeiawww) processBatchProposals(batchProposals www.BatchProposals,
+	user *user.User) (*www.BatchProposalsReply, error) {
+
+	log.Tracef("processBatchProposals")
+
+	props, err := p.getProps(batchProposals.Tokens)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(*props) != len(batchProposals.Tokens) {
+		return nil, www.UserError{
+			ErrorCode: www.ErrorStatusProposalNotFound,
+		}
+	}
+
+	reply := www.BatchProposalsReply{
+		Proposals: *props,
+	}
+
+	for i := range *props {
+		// Vetted proposals are viewable by everyone. The contents of
+		// an unvetted proposal is only viewable by admins and the
+		// proposal author. Unvetted proposal metadata is viewable by
+		// everyone.
+		if reply.Proposals[i].State == www.PropStateUnvetted {
+			var isAuthor bool
+			var isAdmin bool
+			// This is a public route so a user may not exist
+			if user != nil {
+				isAdmin = user.Admin
+				isAuthor = (reply.Proposals[i].UserId == user.ID.String())
+			}
+
+			// Strip the non-public proposal contents if user is
+			// not the author or an admin
+			if !isAuthor && !isAdmin {
+				reply.Proposals[i].Name = ""
+				reply.Proposals[i].Files = make([]www.File, 0)
+			}
+		}
+	}
+
+	return &reply, nil
```

Comment:
The purpose of this route is to populate the proposal list views in the gui.  Proposal files do not need to be returned for this route.